packages/flutter_tools/lib/src/migrations/cmake_policy_migration.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/project_migrator.dart';
+import '../cmake_project.dart';
+
+// Migrate CMake's policy to CMP0135.
+// See https://github.com/flutter/flutter/issues/116866.
+class CmakePolicyMigration extends ProjectMigrator {
+  CmakePolicyMigration(CmakeBasedProject project, super.logger)
+    : _cmakeFile = project.cmakeFile;
+
+  final File _cmakeFile;
+
+  @override
+  void migrate() {
+    if (!_cmakeFile.existsSync()) {
+      logger.printTrace('CMake project not found, skipping cmake_policy() migration');
+      return;
+    }
+
+    final String originalProjectContents = _cmakeFile.readAsStringSync();
+    // Example:
+    //
+    // cmake_policy(SET CMP0063 NEW)
+
+    // Match the whole add_custom_command() and append VERBATIM unless it
+    // already exists.
+    final RegExp cmakePolicy = RegExp(
+      r'^cmake_policy\(SET CMP0063 NEW\)$',
```

Comment:
The comments above the statement are not present in the `CMakeList.txt`s of all first party packages. E.g. `url_launcher_windows` which I have been using to repro the problem does not contain those comments.

Reply:
It looks like the template changed substantially after `cmake_policy(SET CMP0063 NEW)` was introduced. Two potential options:

1. We update this migration to recognize two patterns: the old template used to create first party packages and the current template with comments.
2. We update this migration to recognize just the current template with comments.

I'd lean towards doing the first option if feasible.

Reply:
I think this is no longer relevant. As of https://github.com/flutter/packages/pull/3828, we will not plan for the same kind of migration

Reply:
I'm confused. Why don't we need a migration if we modify the app template? (I'd understand why we wouldn't need a migration if we were only updating the plugin template)

Reply:
@stuartmorgan suggested updating the template going forward without needing a migration on the linked PR. Do you think you can explain the rationale here for clarity?

Reply:
The way I thought this would work when I recommended the migrator was that we could add the new policy setting at the top level, and it would fix everything, so we needed a migrator so that everyone seeing warnings would get fixes.

Since it turns out that this apparently doesn't work (per other comments here), fixing the current set of warnings has to be done in each plugin that is triggering them, and preventing this warning in new plugins is fixed by updating the plugin template.

Even though we can't fix the original issue in the app template, we should update the app template to make it better by setting a `VERSION`-based cmake_policy, since that is:
- recommended by CMake docs (I missed that when doing the initial policy setting),
- more comprehensive, so that new apps will have *all* the new policy behaviors, not just these two.

That's not a pressing concern though, so there's no need to migrate existing projects.

Reply:
Gotcha, thanks for the additional information!